### ISSUE TITLE
suggestion

### ISSUE THREAD

**Issue by khaldounalhalabi** [inFureal/artisan-gui#44]
I am trying to build GUI for my Laravel package but the commands in my package asks questions and the the behavior of the package depends on the input  of those questions now I am searching for a way to resolve this then I saw your package and I think maybe we can cooperate together to add this feature to your package.

any way If I find the way to do that I'll notice you maybe you can add it to your package

**Comment by infureal** [comment]
Hello. 
It’s hard to do such thing because PHP is waiting for std input. There is solution for your issue - make optional arguments in command and if they not present - ask for input.

**Comment by khaldounalhalabi** [comment]
the problem is that the questions are a lot I am trying to do it without that and I consider it as a last option 
as I mentioned before if I have found the solution I'll notice you maybe it helps you add a new feature to your package 
thanks for sharing .

**Comment by infureal** [comment]
There is one think what I thinking about, but it requires child processes and websocket-ish connection

**Comment by khaldounalhalabi** [comment]
I think (not sure) that those requirements need different configuration among the servers .

**Comment by infureal** [comment]
Not at all. Or just drop child process and replace it with one supervisor command. For example: 
- Frontend send encrypted message to peerjs server 
- CLI will receive and decrypt message
- Run command and send chunks of result with asking for promt if needed

**Comment by khaldounalhalabi** [comment]
Oh I see , I think we can stick with making a lot of optional arguments
